File: HackerRank/Algorithms/1. Warmup/Staircase.cpp
#include <iostream>
using namespace std;

int main() {
    /* Enter your code here.
       Read input from STDIN.
       Print output to STDOUT */
    int input;
    int i, j;
    cin >> input;

    for(i = 0; i < input; ++i) {
        for(j = 0; j < input-i-1; ++j) {
            cout << ' ';
        }
        for(; j < input; ++j) {
            cout << '#';
        }
        cout << '\n';
    }
    return 0;
}

File: HackerRank/Algorithms/1. Warmup/Staircase.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here.
           Read input from STDIN.
           Print output to STDOUT.
           Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        int input = scan.nextInt();
        scan.close();

        for(int i = 0; i < input; ++i) {
            int j;
            for(j = 0; j < input-i-1; ++j) {
                System.out.print(' ');
            }
            for(; j < input; ++j) {
                System.out.print('#');
            }
            System.out.print('\n');
        }
    }
}
